### PR TITLE
Make a test farm tests package

### PR DESCRIPTION
**Pull request by bmw**
Fixes https://github.com/certbot/certbot/issues/8781.

This PR makes our test farm tests into a normal package so it and its dependencies can be tracked and installed like our other packages.

Other noteworthy changes in this PR:

* Rather than continuing to place logs in your CWD, they're placed in a temporary directory that is printed to the terminal.
*  `tests/letstest/auto_targets.yaml` was deleted rather than renamed because the file is no longer used.

You can see test farm tests passing with this change at https://dev.azure.com/certbot/certbot/_build/results?buildId=3896&view=results.

### CODE REVIEW

File: tools/pinning/pyproject.toml [hunk 1]
```diff
@@ -32,9 +32,18 @@ certbot-nginx = {path = "../../certbot-nginx", extras = ["docs"]}
...
 
 # Extra dependencies
+# awscli is just listed here as a performance optimization. As of writing this,
+# there are some conflicts in shared dependencies between it and other packages
+# we depend on. To try and resolve them, poetry searches through older versions
+# of awscli to see if that resolves the conflict, but there are over 1000
+# versions of awscli on PyPI so this process takes too long. Providing a high
+# minimum version here prevents poetry from searching this path and it fairly
+# quickly resolves the dependency conflict in another way.
+awscli = "^1.19.62"
```
Comment (adferrand) [inline tools/pinning/pyproject.toml:46]: Wouldn't it be better to use `>=1.19.62` here?

**Review by adferrand — CHANGES_REQUESTED** [review]
Just one question about `awscli` version pinning and we are good!

File: tools/pinning/pyproject.toml (hunk 1 above)
Comment (bmw) [reply tools/pinning/pyproject.toml]: We probably should since that's what we use everywhere else. I made that change.

**Review by adferrand — APPROVED** [review]
This LGTM!

**Pull request merged**